[FIX] l10n_uy_edi: Restore native invisibility of restrict_mode_hash_table for bank and cash journals

fw-bot-adhoc opened the pull request:
Odoo natively sets the field `restrict_mode_hash_table` as invisible for bank and cash journals. However, this behavior was overridden by commit 20c72be367a2e93619c3b862b90c66e0aab331a8, which made the field visible for all journals, potentially leading to configuration errors.

This fix restores the original invisibility for bank and cash journals while keeping the exception for electronic customer invoices in Uruguay (type = sale, l10n_uy_type = electronic).

Forward-Port-Of: ingadhoc/uruguay#369

rov-adhoc committed:
Cherry pick of c37c1895efca93e8666a36e6021c3b3d9cbf4d2f failed

stdout:
CONFLICT (modify/delete): l10n_uy_edi/__manifest__.py deleted in ff37822ad897b8e70d49ed0f2e79411fca201af3 and modified in c37c1895efca93e8666a36e6021c3b3d9cbf4d2f.  Version c37c1895efca93e8666a36e6021c3b3d9cbf4d2f of l10n_uy_edi/__manifest__.py left in tree.
CONFLICT (modify/delete): l10n_uy_edi/views/account_journal_view.xml deleted in ff37822ad897b8e70d49ed0f2e79411fca201af3 and modified in c37c1895efca93e8666a36e6021c3b3d9cbf4d2f.  Version c37c1895efca93e8666a36e6021c3b3d9cbf4d2f of l10n_uy_edi/views/account_journal_view.xml left in tree.

stderr:

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/uruguay/pull/370.png)](https://mergebot.adhoc.ar/ingadhoc/uruguay/pull/370)

fw-bot-adhoc commented:
@rov-adhoc @maq-adhoc cherrypicking of pull request ingadhoc/uruguay#369 failed.

* c37c1895efca93e8666a36e6021c3b3d9cbf4d2f <- on this commit
* aad9b1697dd7e301c631c5be35c6feb925528cec

stdout:
```
CONFLICT (modify/delete): l10n_uy_edi/__manifest__.py deleted in ff37822ad897b8e70d49ed0f2e79411fca201af3 and modified in c37c1895efca93e8666a36e6021c3b3d9cbf4d2f.  Version c37c1895efca93e8666a36e6021c3b3d9cbf4d2f of l10n_uy_edi/__manifest__.py left in tree.
CONFLICT (modify/delete): l10n_uy_edi/views/account_journal_view.xml deleted in ff37822ad897b8e70d49ed0f2e79411fca201af3 and modified in c37c1895efca93e8666a36e6021c3b3d9cbf4d2f.  Version c37c1895efca93e8666a36e6021c3b3d9cbf4d2f of l10n_uy_edi/views/account_journal_view.xml left in tree.

```

Either perform the forward-port manually (and push to this branch, proceeding as usual) or close this PR (maybe?).

In the former case, you may want to edit this PR message as well.

:warning: after resolving this conflict, you will need to merge it via @roboadhoc.

More info at https://github.com/odoo/odoo/wiki/Mergebot#forward-port
